test/hlevel.cooltt

```diff
@@ -55,3 +55,50 @@ def trans (A : type) (p : 𝕀 → A) (q : (i : 𝕀) → sub A {i=0} {p 1}) : p
 
 def contr-prop (A : type) (A/contr : is-contr A) : is-prop A =
   a a' => trans A {symm A {{snd A/contr} a}} {{snd A/contr} a'}
+
+def prop-set (A : type) (A/prop : is-prop A) : is-set A =
+  a b p q i j => hcom A 0 1 {∂ i ∨ ∂ j} {k _ =>
+    [ k=0 ∨ j=0 => A/prop a {p j} k
+    | j=1 => A/prop a {p j} k
+    | i=0 => A/prop a {p j} k
+    | i=1 => A/prop a {q j} k
+    ]
+  }
+
+def raise-hlevel : (l : nat) (A : type) → has-hlevel l A → has-hlevel {suc l} A =
+  let aux : (m : nat) → (B : type) → has-hlevel {suc m} B → has-hlevel {suc {suc m}} B =
+    elim [
+    | zero => prop-set
+    | suc {l => ih} => B B/lvl b b' => ih {path B b b'} {B/lvl b b'}
+    ]
+  in
+  elim [
+  | zero => contr-prop
+  | suc l => aux l
+  ]
+
+def prop-hlevel : (l : nat) (A : type) → is-prop A → has-hlevel {suc l} A =
+  elim [
+  | zero => _ A/prop => A/prop
+  | suc {l => ih} => A A/prop => raise-hlevel {suc l} A {ih A A/prop}
+  ]
+
+def path-hlevel : (l : nat) (A : type) (A/level : has-hlevel {suc l} A) (a : A) (a' : A)
+                 → has-hlevel l {path A a a'} =
+  elim [
+  | zero => A A/prop a a' => [A/prop a a', p => prop-set A A/prop a a' {A/prop a a'} p]
+  | suc l => A A/lvl a a' => A/lvl a a'
+  ]
+
+def path/based/contr (A : type) (a : A) : is-contr {(x : A) × path A a x} =
+  [ [a, i => a]
+  , x i =>
+    let aux : 𝕀 → A = j =>
+      hcom A 0 j {∂ i} {k _ =>
+        [ k=0 ∨ i=0 => a
+        | i=1 => {snd x} k
```

Comment:
I think we should allow `snd x k` here...

Reply:
sounds like a grammar bug!! 🧞‍♂️

Reply:
Sounds like someone just made a Wish? (@clayrat in case you are confused, we now call every grammar proposal as a wish due to #69)

Reply:
Your Wish was Granted. #170

> Sounds like someone just made a Wish? (@clayrat in case you are confused, we now call every grammar proposal as a wish due to #69)

Reply:
Alright, #170 is merged, so we can update this PR and drop the brackets.

Reply:
OK, I have a new PR that makes the change I actually wanted: https://github.com/RedPRL/cooltt/pull/171

Sorry for mis-reading the thread and giving the impression that I wanted this crazy `fst x y z` to work lol. I just wanted `fst x y` to not be parsed as `fst {x y}`.

Reply:
So, just for clarity - the line in question should stay `{snd x} k`?

Reply:
@clayrat Yes that is correct, sorry about that -- I had some kind of Brain Malfunction last night when I agreed to the other proposal lol.